src/components/gcds-site-menu/gcds-site-menu.tsx

```diff
@@ -0,0 +1,310 @@
+        //menuitem.setAttribute("role", "menuitem");
+        // Check if a tag has siblings
+        if (menuitem.parentNode.children.length > 1) {
+          // Apply attributes for sub menus
+          this.setAttributes(menuitem, {"aria-expanded": "false", "aria-haspopup": "true"});
+          // Append sub menu button trigger
+          this.appendSubMenuButton(menuitem);
+          if (menuitem.closest("ul").parentNode.nodeName == "li") {
+            menuitem.setAttribute("tabindex", "-1");
+          }
+        }
+      });
+    }
+
+    // if (this.menuMobileLayout == 'toolbar') {
+    //   if (mainMenus.length > 1) {
+    //     var displayMenu = mainMenus[0];
+    //     var displayMenuChildren = displayMenu.children;
+
+    //     if (displayMenuChildren.length > 2) {
+    //       for (var i = 0; i < displayMenuChildren.length; i++) {
+    //         if (i >1) {
+    //           displayMenuChildren[i].setAttribute('data-toolbar-child', '');
+    //         }
+    //       }
+    //     }
+
+    //     mainMenus[1].setAttribute('data-toolbar-parent', '');
+    //     for (var i =0; i < mainMenus[1].children.length; i++) {
+    //       mainMenus[1].children[i].setAttribute('data-toolbar-child', '');
+    //     }
+    //   }
+    // }
+  }
+
+  async componentWillLoad() {
+    // Define lang attribute
+    if(!this.el.getAttribute("lang")) {
+      if (document.documentElement.getAttribute("lang") == "en" || !document.documentElement.getAttribute("lang")) {
+        this.lang = "en";
+      } else {
+        this.lang = "fr";
+      }
+    } else if(this.el.getAttribute("lang") == "en") {
+      this.lang = "en";
+    } else {
+      this.lang = "fr";
+    }
+
+    this.validateDesktopLayout(this.menuDesktopLayout);
+    this.validateMobileLayout(this.menuMobileLayout);
+
+
+    // Add required attributes to slotted <ul>
+    await this.configureMenu();
+
+  }
+
+  componentDidLoad() {
+    var menus = [];
+    // Add ULs to shadow DOM
+    var elementChildren = this.el.children;
+    // Loop through slotted elements
+    for (var i = 0; i < elementChildren.length; i++) {
+      // Act only on ul
+      if (elementChildren[i].nodeName == "UL") {
+        menus = menus.concat(elementChildren[i]);
+      }
+    }
+
+    for (var i = 0; i < menus.length; i++) {
+      var container = this.el.shadowRoot.querySelector("[data-h2-menu-container]");
+      container.insertBefore(menus[i], container.querySelector('[data-optional-right]'));
+    }
+
+    if (this.menuDesktopLayout == "sidebar") {
+      // this.el.shadowRoot.querySelector("nav").style.setProperty(`--page-top-offset`, `${this.el.offsetTop}px`);
+    }
+
+    var hostElement = this.el;
+    var mobileLayout = this.menuMobileLayout;
+    const mediaQuery = window.matchMedia('screen and (min-width: 64em)');
+
+    // Check if loaded in mobile size
+    if (!mediaQuery.matches && mobileLayout == "drawer") {
+      document.querySelector("body").style.paddingBottom = "3rem";
```

Comment:
Can we make this change in CSS instead?

Reply:
To allow space for the mobile menu button, the style had to be added directly to the body since the encapsulated CSS from the component wouldn't be able to change the body element.